scrapers/src/catalog-scraper/reqGroup_scraper.ts

```diff
@@ -202,6 +219,118 @@ function processAndSection(
   }
 }
 
+function createAndReqList(
+  $: CheerioStatic,
+  rows: CheerioElement[]
+): Requirement[] {
+  let seperate_sections: Map<
+    CheerioElement[],
+    [SubSectionType, number]
+  > = splitRows($, rows);
+  let reqList: Requirement[] = [];
+  Array.from(seperate_sections.entries()).forEach(
+    ([sectionRows, [type, credits]]) => {
+      let requirements = [];
+      switch (+type) {
+        case SubSectionType.COURSES:
+          requirements = evaluateRows($, sectionRows);
+          reqList = reqList.concat(requirements);
+          break;
+        case SubSectionType.CREDIT:
+          requirements = evaluateRows($, sectionRows);
+          if (requirements.length !== 0) {
+            reqList.push({
+              type: "CREDITS",
+              minCredits: credits,
+              maxCredits: credits,
+              courses: requirements,
+            });
+          }
+          break;
+        case SubSectionType.RANGE:
+          let range = processCourseRange($, sectionRows, credits);
+          reqList.push(range);
+          break;
+        default:
+          break;
+      }
+    }
+  );
+  return reqList;
+}
+
+function evaluateRows($: CheerioStatic, rows: CheerioElement[]) {
+  let reqList: Requirement[] = [];
+  rows.forEach((row: CheerioElement) => {
+    let requirement: Requirement | undefined = parseRowAsRequirement($, row);
+
+    if (requirement && isRequirement(requirement)) {
+      if (isOrRow($, row)) {
+        let orVal = reqList.pop();
+        if (orVal) {
+          requirement = createIOrCourse(requirement, orVal);
+        }
+      }
+      reqList.push(requirement);
+    }
+  });
+  return reqList;
+}
+
+function splitRows(
+  $: CheerioStatic,
+  rows: CheerioElement[]
+): Map<CheerioElement[], [SubSectionType, number]> {
+  let split = new Map<CheerioElement[], [SubSectionType, number]>();
+  let sectionType = SubSectionType.COURSES;
+  let credits = 0;
+  let rowCollection: CheerioElement[] = [];
+  rows.forEach((row: CheerioElement) => {
+    let comment = $(row)
+      .find("span.courselistcomment")
+      .text();
+    if (comment.length > 0) {
+      if (ORTagMap.hasOwnProperty(comment)) {
```

Comment:
The nested ifs could be helper methods, that would make this function a lot more readable

Reply:
I tried it out and it required a function with 7 arguments, which seemed like too much to make it worth it